tests/link/pytorch/test_basic.py

```diff
@@ -0,0 +1,205 @@
+from collections.abc import Callable, Iterable
+from functools import partial
+
+import numpy as np
+import pytest
+
+from pytensor.compile.function import function
+from pytensor.compile.mode import get_mode
+from pytensor.compile.sharedvalue import SharedVariable, shared
+from pytensor.configdefaults import config
+from pytensor.graph.basic import Apply
+from pytensor.graph.fg import FunctionGraph
+from pytensor.graph.op import Op
+from pytensor.raise_op import CheckAndRaise
+from pytensor.tensor.type import scalar, vector
+
+
+torch = pytest.importorskip("torch")
+
+
+pytorch_mode = get_mode("PYTORCH")
+py_mode = get_mode("FAST_COMPILE")
+
+
+def compare_pytorch_and_py(
+    fgraph: FunctionGraph,
+    test_inputs: Iterable,
+    assert_fn: Callable | None = None,
+    must_be_device_array: bool = True,
+    pytorch_mode=pytorch_mode,
+    py_mode=py_mode,
+):
+    """Function to compare python graph output and pytorch compiled output for testing equality
+
+    Parameters
+    ----------
+    fgraph: FunctionGraph
+        PyTensor function Graph object
+    test_inputs: iter
+        Numerical inputs for testing the function graph
+    assert_fn: func, opt
+        Assert function used to check for equality between python and pytorch. If not
+        provided uses np.testing.assert_allclose
+    must_be_device_array: Bool
+        Checks if torch.device.type is cuda
+
+
+    """
+    if assert_fn is None:
+        assert_fn = partial(np.testing.assert_allclose)
+
+    fn_inputs = [i for i in fgraph.inputs if not isinstance(i, SharedVariable)]
+
+    pytensor_torch_fn = function(fn_inputs, fgraph.outputs, mode=pytorch_mode)
+    pytorch_res = pytensor_torch_fn(*test_inputs)
+
+    if must_be_device_array:
+        if isinstance(pytorch_res, list):
+            assert all(isinstance(res, torch.Tensor) for res in pytorch_res)
+        else:
+            assert pytorch_res.device.type == "cuda"
+
+    pytensor_py_fn = function(fn_inputs, fgraph.outputs, mode=py_mode)
+    py_res = pytensor_py_fn(*test_inputs)
+
+    if len(fgraph.outputs) > 1:
+        for j, p in zip(pytorch_res, py_res):
+            assert_fn(j.cpu(), p)
+    else:
+        assert_fn([pytorch_res[0].cpu()], py_res)
+
+    return pytensor_torch_fn, pytorch_res
+
+
+@pytest.mark.parametrize("device", ["cpu", "cuda"])
+def test_pytorch_FunctionGraph_once(device):
+    if torch.cuda.is_available() is False:
```

Comment:
Only want to skip on the GPU parametrization
```suggestion
    if device == "cuda" and not torch.cuda.is_available():
```

Same thing in the other tests